daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionVersion.scala

```diff
@@ -153,7 +153,7 @@ private[lf] object TransactionVersions
     Either.cond(
       !(supportedTxVersions.max precedes transactionVersion),
       transactionVersion,
-      s"inferred version $transactionVersion is not supported"
+      s"inferred transaction version ${transactionVersion.protoValue} is not allowed"
```

Comment:
Can we use proper types to represent these errors instead of strings?

Reply:
In another PR.